folium/map.py

```diff
@@ -335,6 +348,140 @@ def render(self, **kwargs):
+                        {% if this.toLocaleString %}.toLocaleString(){% endif %} }`
+                    ).join('<br>'))+'</div>'
+                {% elif this.text %}
+                    return '<div{% if this.style %} style="{{this.style}}"{% endif%}>'
+                    + '{{ this.text }}'+'</div>'
+                {% else %}
+                    return '<div{%if this.style%} style="{{this.style}}"{%endif%}>'+'{{ this.__str__() }}'+'</div>'
+                {% endif %}
+                }{% if this.kwargs %}, {{ this.kwargs }}{% endif %});
+            {% endmacro %}
+        """)
+
+    def __init__(self, text=None, fields=None, aliases=None, labels=True,
+                 toLocaleString=False, style=None, **kwargs):
+        super(Tooltip, self).__init__()
+        self._name = "Tooltip"
+
+        if fields:
+            assert isinstance(fields, (list, tuple)), "Please pass a list or " \
+                                                      "tuple to fields."
+        if all((fields, aliases)):
+            assert isinstance(aliases, (list, tuple))
+            assert len(fields) == len(aliases), "fields and aliases must have" \
+                                                " the same length."
+        assert isinstance(labels, bool), "labels requires a boolean value."
+        assert not all((fields, text)), "Please choose either fields or text."
+        assert any((fields, text)), "Please choose either fields or text."
+        assert isinstance(toLocaleString, bool), "toLocaleString must be " \
+                                                 "boolean."
+        self.valid_kwargs = {"pane": (str,),
+                             "offset": (tuple,),
+                             "direction": (str,),
+                             "permanent": (bool,),
+                             "sticky": (bool,),
+                             "interactive": (bool,),
+                             "opacity": (float, int)}
+        if kwargs:
+            for key in kwargs.keys():
+                assert key in self.valid_kwargs.keys(), "The key {0} was not " \
+                                                        "available in the " \
+                                                        "keys: {1}".format(
+                    key, ', '.join(self.valid_kwargs.keys()))
+                assert isinstance(kwargs[key], self.valid_kwargs[key]), \
+                    "{0} must be of the following " \
+                    "types: {1}".format(key, self.valid_kwargs[key])
+            self.kwargs = json.dumps(kwargs)
+        self.fields = fields
+        self.aliases = aliases
+        self.text = text.__str__()
+        self.labels = labels
+        self.toLocaleString = toLocaleString
+        if style:
+            assert isinstance(style, str), \
+                "Pass a valid inline HTML style property string to style."
+            # noqa outside of type checking.
+            self.style = style
+        if self.fields:
```

Comment:
This can become a oneliner:

`self.result = self.fields if self.fields else self.text`

Reply:
Looking through it, it seems like `self.result` was a relic from when this was in the GeoJson template. It can go now.